src/dma/mod.rs

```diff
@@ -185,21 +283,38 @@ pub enum CurrentBuffer {
     /// The first buffer (m0ar) is in use.
     FirstBuffer,
     /// The second buffer (m1ar) is in use.
-    DoubleBuffer,
+    SecondBuffer,
 }
 
 impl Not for CurrentBuffer {
     type Output = CurrentBuffer;
 
     fn not(self) -> Self::Output {
         if self == CurrentBuffer::FirstBuffer {
-            CurrentBuffer::DoubleBuffer
+            CurrentBuffer::SecondBuffer
         } else {
             CurrentBuffer::FirstBuffer
         }
     }
 }
 
+/// Structure to get or set common interrupts setup
+pub struct DmaCommonInterrupts {
+    pub transfer_complete: bool,
+    pub half_transfer: bool,
+    pub transfer_error: bool,
+    pub direct_mode_error: bool,
+}
+
+/// Structure returned by Stream or Transfer all_flags() method.
+pub struct DmaFlags {
+    pub transfer_complete: bool,
+    pub half_transfer: bool,
+    pub transfer_error: bool,
+    pub direct_mode_error: bool,
+    pub fifo_error: bool,
+}
```

Comment:
```suggestion
pub struct DmaFlags {
    pub TransferComplete: bool,
    pub HalfTransfer: bool,
    pub TransferError: bool,
    pub DirectModeError: bool,
    pub FifoError: bool,
}
```
Same for `DmaCommonInterrupts`

Reply:
Even better to use bitflags here:
https://github.com/YruamaLairba/stm32f4xx-hal/pull/1

Reply:
Why ? Doing that will trigger a warning!

Reply:
oh. sorry. this is field